### ISSUE TITLE
Incorrect Conv2DTranspose behavior on CPU

### ISSUE THREAD

**Issue by azai91** [apache/incubator-mxnet#13876]
The native CPU implementation of Conv2DTranspose produces incorrect results.
```
import mxnet as mx
data = mx.nd.array(((0,0,0),
                    (0,1,0),
                    (0,0,0)))
kernel = mx.nd.array(((1,2,3),
                      (4,5,6),
                      (7,8,9)))
data_batch = data.expand_dims(0).expand_dims(0)
weight = kernel.expand_dims(0).expand_dims(0)
# initialize and set weight
conv = mx.gluon.nn.Conv2DTranspose(in_channels=1, channels=1,
                                   kernel_size=(3,3), padding=(2,2),
                                   strides=(2,2), dilation=(2,2))
conv.initialize()
conv.weight.set_data(weight)
print(conv(data_batch))
```

```
[[[[0. 0. 0. 0. 0.]
   [0. 1. 0. 2. 3.]
   [0. 0. 0. 0. 0.]
   [0. 4. 0. 5. 6.]
   [0. 7. 0. 8. 9.]]]]
<NDArray 1x1x5x5 @cpu(0)>
```

The output should be
```
[[[[1. 0. 2. 0. 3.]
   [0. 0. 0. 0. 0.]
   [4. 0. 5. 0. 6.]
   [0. 0. 0. 0. 0.]
   [7. 0. 8. 0. 9.]]]]
<NDArray 1x1x5x5 @cpu(0)>
```

## Description
(Brief description of the problem in no more than 2 sentences.)

## Environment info (Required)

```
What to do:
1. Download the diagnosis script from https://raw.githubusercontent.com/apache/incubator-mxnet/master/tools/diagnose.py
2. Run the script using `python diagnose.py` and paste its output here.

```

Package used (Python/R/Scala/Julia):
(I'm using ...)

For Scala user, please provide:
1. Java version: (`java -version`)
2. Maven version: (`mvn -version`)
3. Scala runtime if applicable: (`scala -version`)

For R user, please provide R `sessionInfo()`:

## Build info (Required if built from source)

Compiler (gcc/clang/mingw/visual studio):

MXNet commit hash:
(Paste the output of `git rev-parse HEAD` here.)

Build config:
(Paste the content of config.mk, or the build command.)

## Error Message:
(Paste the complete error message, including stack trace.)

## Minimum reproducible example
(If you are using your own code, please provide a short script that reproduces the error. Otherwise, please provide link to the existing example.)

## Steps to reproduce
(Paste the commands you ran that produced the error.)

1. See above script

**Comment by azai91** [comment]
@apeforest @Vikas89 tracking issue here

**Comment by piyushghai** [comment]
@azai91 Thank you for submitting the issue! I'm labeling it so the MXNet community members can help resolve it.
@mxnet-label-bot add [Bug, Operator]

**Comment by hetong007** [comment]
This is identical to https://github.com/apache/incubator-mxnet/issues/11203

**Comment by azai91** [comment]
got it. closing this ticket since it is a duplicate.
